### PR TITLE
Make OSVersion lazy

### PR DESCRIPTION
**Pull request by BrennanConroy**
@pranavkm

### CODE REVIEW

File: src/Microsoft.Extensions.RuntimeEnvironment.Sources/RuntimeEnvironment.cs [hunk 1]
```diff
@@ -9,15 +9,21 @@ internal static class RuntimeEnvironment
...
             RuntimeType = GetRuntimeType();
         }
 
+        private static readonly Lazy<string> _operatingSystemVersion = new Lazy<string>(() => Native.PlatformApis.GetOSVersion());
+
         /// <summary>
         /// Don't use in shipping packages
         /// </summary>
-        static public string OperatingSystemVersion { get; }
+        static public string OperatingSystemVersion
```
Comment (pranavkm) [inline src/Microsoft.Extensions.RuntimeEnvironment.Sources/RuntimeEnvironment.cs]: `public static string OperatingSystemVersion => _operatingSystemVersion.Value;`

File: src/Microsoft.Extensions.RuntimeEnvironment.Sources/RuntimeEnvironment.cs [hunk 2]
```diff
@@ -9,15 +9,21 @@ internal static class RuntimeEnvironment
     {
         static RuntimeEnvironment()
         {
-            OperatingSystemVersion = Native.PlatformApis.GetOSVersion();
-
             RuntimeType = GetRuntimeType();
         }
 
+        private static readonly Lazy<string> _operatingSystemVersion = new Lazy<string>(() => Native.PlatformApis.GetOSVersion());
```
Comment (pranavkm) [inline src/Microsoft.Extensions.RuntimeEnvironment.Sources/RuntimeEnvironment.cs]: Above the ctor?

File: src/Microsoft.Extensions.RuntimeEnvironment.Sources/RuntimeEnvironment.cs [hunk 3]
```diff
@@ -9,15 +9,21 @@ internal static class RuntimeEnvironment
     {
         static RuntimeEnvironment()
         {
-            OperatingSystemVersion = Native.PlatformApis.GetOSVersion();
-
             RuntimeType = GetRuntimeType();
```
Comment (pranavkm) [inline src/Microsoft.Extensions.RuntimeEnvironment.Sources/RuntimeEnvironment.cs]: Maybe move this to to the property too 
`static public string RuntimeType { get; } = GetRuntimeType();` and you can remove the static ctor.

**Comment by pranavkm** [comment]
:shipit:

**Comment by BrennanConroy** [comment]
Looks so beautiful now!

**Pull request merged**